app/controllers/api/v1/items_controller.rb

```diff
@@ -1,6 +1,12 @@
 class Api::V1::ItemsController < ApplicationController
   def index
-    render json: ItemSerializer.new(Item.all)
+    if (params[:merchant_id])
+      merchant = Merchant.find(params[:merchant_id])
+      @items = merchant.items
+    else
+      @items = Item.all
+    end
```

Comment:
this handles whether to show all items, or only those belonging to a given merchant